src/consensus/consensus.rs

```diff
@@ -32,7 +38,7 @@ pub struct Config {
     pub block_time: Duration,
 
     pub max_messages_per_block: u32,
-    pub validator_addresses: Vec<String>,
```

Comment:
Let's keep the existing config for backwards compatibility. I think we'd need to bring back the `validator_set_for` method below, and have it check if validator_addresses is present first and construct ValidatorSetConfig from it.

Reply:
Hmm, that seems like it would complicate things a lot, would it not? Then both of the options have to be optional and we would need handling for if one or the other or both are defined? What takes precedence if both are specified for instance 😄

Reply:
It's fine to panic if both are defined, that should not happen. This is mainly to make migration to the new config easier. If the logic is too complex, then I think we can migrate all existing configs, but that makes deployment more complicated. But I think it's easier in code? What do you think?

Reply:
Ended up just defaulting to the new format if both are defined and adding backwards compat in https://github.com/farcasterxyz/snapchain/pull/279/commits/b7f04b5c674698afe47f34029a058e91996e578f